tests/test_buffer.py

```diff
@@ -0,0 +1,215 @@
+        self.app_db.wait_for_field_match("BUFFER_PG_TABLE", "Ethernet0:3-4", {"profile": "[BUFFER_PROFILE_TABLE:" + expectedProfile + "]"})
+
+        # remove lossless PG 3-4 on interface
+        dvs.runcmd("config interface buffer priority-group lossless remove Ethernet0 3-4")
+
+    @pytest.mark.skip(reason="Traditional buffer model runs on VS image currently.")
+    def test_changeCableLen(self, dvs, testlog):
+        self.setup_db(dvs)
+
+        # configure lossless PG 3-4 on interface
+        dvs.runcmd("config interface buffer priority-group lossless add Ethernet0 3-4")
+
+        # Change to new cable length
+        dvs.runcmd("config interface cable-length Ethernet0 " + self.cableLenTest1)
+        expectedProfile = self.make_lossless_profile_name(self.originalSpeed, self.cableLenTest1)
+        self.app_db.wait_for_entry("BUFFER_PROFILE_TABLE", expectedProfile)
+        self.app_db.wait_for_field_match("BUFFER_PG_TABLE", "Ethernet0:3-4", {"profile": "[BUFFER_PROFILE_TABLE:" + expectedProfile + "]"})
+
+        # Remove the lossless PGs
+        dvs.runcmd("config interface buffer priority-group lossless remove Ethernet0")
+        self.app_db.wait_for_deleted_entry("BUFFER_PG_TABLE", "Ethernet0:3-4")
+
+        # Change to another cable length
+        dvs.runcmd("config interface cable-length Ethernet0 " + self.cableLenTest2)
+        # Check whether the old profile has been removed
+        self.app_db.wait_for_deleted_entry("BUFFER_PROFILE_TABLE", expectedProfile)
+
+        # Re-add lossless PGs
+        dvs.runcmd("config interface buffer priority-group lossless add Ethernet0 3-4")
+        expectedProfile = self.make_lossless_profile_name(self.originalSpeed, self.cableLenTest2)
+        # Check the BUFFER_PROFILE_TABLE and BUFFER_PG_TABLE
+        self.app_db.wait_for_entry("BUFFER_PROFILE_TABLE", expectedProfile)
+        self.app_db.wait_for_field_match("BUFFER_PG_TABLE", "Ethernet0:3-4", {"profile": "[BUFFER_PROFILE_TABLE:" + expectedProfile + "]"})
+
+        # Revert the cable length
+        dvs.runcmd("config interface cable-length Ethernet0 " + self.originalCableLen)
+        # Check the BUFFER_PROFILE_TABLE and BUFFER_PG_TABLE
+        self.app_db.wait_for_deleted_entry("BUFFER_PROFILE_TABLE", expectedProfile)
+        expectedProfile = self.make_lossless_profile_name(self.originalSpeed, self.originalCableLen)
+        self.app_db.wait_for_field_match("BUFFER_PG_TABLE", "Ethernet0:3-4", {"profile": "[BUFFER_PROFILE_TABLE:" + expectedProfile + "]"})
+
+        # remove lossless PG 3-4 on interface
+        dvs.runcmd("config interface buffer priority-group lossless remove Ethernet0 3-4")
+
+    @pytest.mark.skip(reason="Traditional buffer model runs on VS image currently.")
+    def test_MultipleLosslessPg(self, dvs, testlog):
+        self.setup_db(dvs)
+
+        # configure lossless PG 3-4 on interface
+        dvs.runcmd("config interface buffer priority-group lossless add Ethernet0 3-4")
```

Comment:
Could you add a negative test with multiple lossless pools with size more than MMU size?

Reply:
By default, only one lossless pool is supported. dynamic buffer calculation assumes only `ingress_lossless_pool` is used.
For the negative test, in general, I think we can verify it in the following steps:
- Get the `mmusize` from `STATE_DB`
- Hack the plugin which is responsible for calculating the pool size, making it return a super large value, which requires
  - prepare a hacked version of the plugin
  - copy it to the DUT and replace the original one with the new one in the swss docker
  - reload config to make the hacked plugin loaded
- Check whether the large pool size is deployed
- Clean up: remove the hacked plugin

As it isn't a simple test, I suggest adding this test in the future.